File: js/programmers/level.1/prgm_12943_1.js
function solution(num) {
  if (num === 1) {
    return 0;
  } else {
    let count = 0;

    while (num !== 1 && count !== 500) {
      num % 2 === 0 ? (num /= 2) : (num = num * 3 + 1);
      count++;
    }
    return num === 1 ? count : -1;
  }
}

File: python/programmers/level.1/prgm_12943_1.py
def solution(n):
    count = 0

    while n != 1:
        if count < 500:
            if n % 2 == 0:
                n /= 2
                count += 1
            else:
                n = n*3 + 1
                count += 1
        else:
            return -1

    return count
